File: Ejercicio_0016/Potenciation.js
function potencia(x, y) {
    let result = x;
    if (x == 0) {
        if (y == 0) {
            result = 1;
        } else {
            result = 0;
        }
    } else if (y == 0) {
        result = 0;
    } else {
        let i = 1;
        while (i < y) {
            result *= x;
            i++;
        }
    }
    alert(`
      ${y}
    ${x}        es igual a ${result}
    `);
}

let num1 = parseInt(prompt("\nDame el numero de la base de la potencia:\n"));

while (num1 < 0) {
    num1 = parseInt(prompt("\nSolo numero natural para base de la potencia:\n"));
}

let num2 = parseInt(prompt("\nDame el numero de la potencia:\n"));

while (num2 < 0) {
    num2 = parseInt(prompt("\nSolo numero natural para de la potencia:\n"));
}

potencia(num1, num2)

File: Ejercicio_0016/Potenciation.java
package Ejercicio_0016;

import javax.swing.JOptionPane;

public class Potenciation {
    public static void main(String[] args) {
        int num1 = Integer.parseInt(JOptionPane.showInputDialog("\nDame el numero de la base de la potencia:\n\n"));

        while (num1 < 0) {
            num1 = Integer.parseInt(JOptionPane.showInputDialog("\nSolo numero natural para base de la potencia:\n\n"));
        }

        int num2 = Integer.parseInt(JOptionPane.showInputDialog("\nDame el numero de la potencia:\n\n"));

        while (num2 < 0) {
            num2 = Integer.parseInt(JOptionPane.showInputDialog("\nSolo numero natural para de la potencia:  \n\n"));
        }

        int result = num1;

        if (num1 == 0) {
            if (num2 == 0) {
                result = 1;
            } else {
                result = 0;
            }
        } else if (num2 == 0) {
            result = 1;
        } else {
            int i = 1;
            while (i < num2) {
                result *= num1;
                i +=1;
            }
        }

        String mensaje = String.format("""
          %d    
        %d      es igual a %d

        """, num2, num1, result);

        JOptionPane.showMessageDialog(null, mensaje);
    }
}